.auto-release.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+repo_slug="mimblewimble/grin"
+token="$GITHUB_TOKEN"
+
+tagname=`git describe --tags --exact-match 2>/dev/null || git symbolic-ref -q --short HEAD`
+
+echo 'make a tarball for the release binary...\n'
+
+if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
+
+    # Do some custom requirements on OS X
+    cd target/release ; rm -f *.tgz; tar zcf "grin-$tagname-$TRAVIS_JOB_ID-osx.tgz" grin
+    /bin/ls -ls *.tgz  | awk '{print $6,$7,$8,$9,$10}'
+    md5 "grin-$tagname-$TRAVIS_JOB_ID-osx.tgz" > "grin-$tagname-$TRAVIS_JOB_ID-osx.tgz"-md5sum.txt
+    /bin/ls -ls *-md5sum.txt  | awk '{print $6,$7,$8,$9,$10}'
+    cd - > /dev/null;
+    echo "tarball generated\n"
+
+    # Only generate changelog on Linux platform, to avoid duplication
+    exit 0
+else
+    # Do some custom requirements on Linux
+    cd target/release ; rm -f *.tgz; tar zcf "grin-$tagname-$TRAVIS_JOB_ID-linux-amd64.tgz" grin
+    /bin/ls -ls *.tgz  | awk '{print $6,$7,$8,$9,$10}'
+    md5sum "grin-$tagname-$TRAVIS_JOB_ID-linux-amd64.tgz" > "grin-$tagname-$TRAVIS_JOB_ID-linux-amd64.tgz"-md5sum.txt
+    /bin/ls -ls *-md5sum.txt  | awk '{print $6,$7,$8,$9,$10}'
+    cd - > /dev/null;
+    echo "tarball generated\n"
+fi
+
+version="$tagname"
+branch="`git symbolic-ref -q --short HEAD`"
+
+# automatic changelog generator
+gem install github_changelog_generator
```

Comment:
I guess it makes sense to use the CI to release the binaries but has it been thought through whether the release process should depend on travis as opposed to a script that can run both in travis and locally?

Reply:
You wouldn't use travis at all when running locally, you would push releases directly in github.

Reply:
Thanks @kargakis for the review~
I'm afraid no, because we can't push directly in github 😄   It's a rule here any change in repo need use PR instead of directly pushing, for better quality control and review.